external/fv3fit/fv3fit/emulation/data/load.py

```diff
@@ -0,0 +1,52 @@
+import logging
+import os
+import tensorflow as tf
+from typing import Callable, List, Sequence
+
+from vcm import get_fs
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_nc_files(path: str) -> List[str]:
+    """
+    Get a list of netCDF files from a remote/local directory
+
+    Args:
+        path: Local or remote gcs path to netCDF directory
+    """
+
+    fs = get_fs(path)
+    files = list(fs.glob(os.path.join(path, "*.nc")))
+
+    return files
+
+
+def batched_to_tf_dataset(
```

Comment:
This function does a few too many things. It transforms a sequence into a tf Dataset, than maps over this dataset, and sets a pretuning strategy. It feels like this function should just be copied verbatim to wherever it is used since it is very abstract and in all the common cases the input data are actually a list of file names? Does this not work?
```
tf.data.Dataset.from_tensor_slices(filenames).flat_map(transform).prefetch(tf.data.AUTOTUNE)
```

The name is also quite specific name given how abstract the function is. This function will work with a very broad class of inputs, not just "batched_sources". Also it mostly is a mapping operation so should have the word "map".

Reply:
> It feels like this function should just be copied verbatim to wherever it is used since it is very abstract and in all the common cases the input data are actually a list of file names? Does this not work?

Yeah, it doesn't work out with the one-liner since the `transform` is not a tensorflow-native operation and it tries to generate graph ops for everything in the provided transform.  I tried hiding the transform in a sequence-like structure using `loaders.batches._sequences.Map`, but that didn't work either.  We could wrap transform in a `tf.py_function`, but that doesn't handle nested structure outputs (i.e., two groups of tuples) natively from what I could tell.  I settled on using the `from_generator` because it seemed the most straightforward to understand.

I made this function a bit more flexible because I was thinking there still might be cases where I use the sequences created by loaders (e.g., non-subsampled data) instead of an input list of netCDF files.

I also had the one-liner convenience function going from files to a tf dataset, but it felt like something I'd put in the place where the training workflow is run.  Also, it wasn't clear what level of convenience I should provide for the transform portion, such as the option to initialize from a config file or a dict if provided.  Would be fine throwing this back in somewhere if you'd find it useful.

>This function will work with a very broad class of inputs, not just "batched_sources". Also it mostly is a mapping operation so should have the word "map".

Maybe `sequence_map_to_tf_dataset`?

> This function does a few too many things.

I can let the user decide whether or not they want to use the pre-fetch operation.  Since we can't generally parallelize when using a generator or py_funcs with map, I figured adding this by default would at least give users the performance boost for tensorflow training.